app/models/database.py

```diff
@@ -73,11 +73,39 @@ def check_if_key_exists(self, key_value, is_user=False) -> bool:
     # READ
 
     def get_link(self, short_url: str) -> Link:
-        pass
         """
-        this function will take a shortlink as a key, and return the link object as the value
+        this function will take a shortlink as a key, and return the link
         """
+        try:
+            #look for the key
+            res = self.table.get_item(Key={'short_url': short_url})
+            if "Item" in res:
+                #Since the url exists, and this is a redirect request, increment the counter
+                self.increment_click(short_url)
+
+                #return the original url for forwarding
+                return res['Item']["original_url"]
+
+        except Exception as e:
+            raise Exception(f"Failed to retrieve link {e}")
+
+
+
 
+    def increment_click(self, short_url):
+
+        #Increment the click counter for a given short URL, the link should be verified to exist already.
+
+        try:
+            self.table.update_item(
+                Key={'short_url': short_url},
+                UpdateExpression='SET clicks = clicks + :val',
```

Comment:
This looks very sketchy and unintuitive. 

Here's what we need to do.

First, add typing to the function definition.

The format should of the update should be something like

1. get the Link
2. Increment the count
3. update the database with that link.

You don't want to use the raw query strings. It doesn't look like to code and leaves you potentially vulnerable to injection attacks. https://www.ibm.com/docs/en/snips/4.6.0?topic=categories-injection-attacks